segmentation_models_pytorch/encoders/mobilenet.py

```diff
@@ -25,10 +25,277 @@
+
+        # building mobile blocks
+        for k, exp, c, se, nl, s in mobile_setting:
+            output_channel = make_divisible(c * width_mult)
+            exp_channel = make_divisible(exp * width_mult)
+            self.features.append(MobileBottleneck(
+                input_channel, output_channel, k, s, exp_channel, se, nl))
+            input_channel = output_channel
+
+        # building last several layers
+        if mode == "large":
+            last_conv = make_divisible(960 * width_mult)
+        elif mode == "small":
+            last_conv = make_divisible(576 * width_mult)
+            # self.features.append(SEModule(last_conv))  # refer to paper Table2, but I think this is a mistake
+        else:
+            raise NotImplementedError
+        # classifier head
+        self.features.append(conv_1x1_bn(
+            input_channel, last_conv, nlin_layer=Hswish))
+        self.features.append(nn.AdaptiveAvgPool2d(1))
+        self.features.append(nn.Conv2d(last_conv, last_channel, 1, 1, 0))
+        self.features.append(Hswish(inplace=True))
+        # make it nn.Sequential
+        self.features = nn.Sequential(*self.features)
+
+        # building classifier
+        self.classifier = nn.Sequential(
+            nn.Dropout(p=dropout),    # refer to paper section 6
+            nn.Linear(last_channel, n_class),
+        )
+
+        self._initialize_weights()
+
+    def forward(self, x):
+        x = self.features(x)
+        x = x.mean(3).mean(2)
+        x = self.classifier(x)
+        return x
+
+    def _initialize_weights(self):
+        # weight initialization
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight, mode="fan_out")
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.ones_(m.weight)
+                nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.Linear):
+                nn.init.normal_(m.weight, 0, 0.01)
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+
+
+class MobileNetV3Encoder(MobileNetV3, EncoderMixin):
+    def __init__(self, depth=5, mode="small", width_mult=1.0, **kwargs):
+        super().__init__(mode=mode, width_mult=width_mult, **kwargs)
+        self._depth = depth
+        self.mode = mode
+        if mode == "small":
+            self._out_channels = (3*width_mult, 16*width_mult, 16*width_mult,
+                                  24*width_mult, 48*width_mult, 96*width_mult)
+            self._out_channels = tuple(map(make_divisible, self._out_channels))
+        elif mode == "large":
+            self._out_channels = (3*width_mult, 16*width_mult, 24*width_mult,
+                                  40*width_mult, 112*width_mult, 160*width_mult)
+            self._out_channels = tuple(map(make_divisible, self._out_channels))
+        else:
+            raise NameError
```

Comment:
raise ValueError("MobileNetV3 mode should be `small` or `large`, got {}".format(mode))